env:
  global:
  - secure: Od2ZYUhrfms2EKvWZ+fhiy2aUMKO5v8km9gvRpLs3fQx/Wo23oQwVVx1lNE/bIZ9Uea596d/Y5VnSkZ4r7HRhwyyMw7F2M+5wbW+RIfHN05XaDma3H9ulF94mWQiMrbT9yFur3AeKb+Wwavl6U0bH7/DsFcXLq/z+1GaRNH/Bnc=

language: python
python:
  - "2.6"
  - "2.7"
  - "3.2"
  - "3.3"

sudo: required
addons:
  apt:
    packages:
      - libgtest-dev

script:
  # NOTE: we must do all testing on the installed python package, not 
  # on the build tree. Otherwise the testing is invalid and may not
  # indicate the code actually works
  #
  # Set pythonpath

  # install
  - cd pymavlink
  - python setup.py build install
  - cd ..

  # Generate messages
  - mavgen.py --lang='Python' 	--output=/tmp/ message_definitions/v1.0/common.xml
  - mavgen.py --lang='C' 	--output=/tmp/ message_definitions/v1.0/common.xml
  - mavgen.py --lang='CS' 	--output=/tmp/ message_definitions/v1.0/common.xml
  - mavgen.py --lang='WLua' 	--output=/tmp/ message_definitions/v1.0/common.xml
  - mavgen.py --lang='Java'	--output=/tmp/ message_definitions/v1.0/common.xml
  - mavgen.py --lang='C'        --wire-protocol='2.0' --output=/tmp/v2 message_definitions/v1.0/common.xml
  - mavgen.py --lang='C++11'    --wire-protocol='2.0' --output=/tmp/v2 message_definitions/v1.0/common.xml
  # Avoid `spurious errors` caused by ~/.npm permission issues
  # ref: https://github.com/travis-ci/travis-ci/issues/2244
  # ref: https://github.com/npm/npm/issues/4815
  # Does it already exist? Who owns? What permissions?
  - ls -lah ~/.npm || mkdir ~/.npm
  # Make sure we own it
  # $USER references the current user in Travis env
  - sudo chown -R $USER ~/.npm
  - "cd pymavlink/generator/javascript && npm test"

  # Test quaternions
  - cd $TRAVIS_BUILD_DIR
  - pymavlink/tools/quaterniontest.py

  # Test C++11 and C interop: require libgtest-dev and at least g++ 4.8
  - make -C pymavlink/generator/CPP11/test/posix run_tests CXX=clang++ GENDIR=/tmp/v2

after_success:
- ./scripts/travis_update_generated_repos.sh
